src/foam/apploader/ModelFileDAO.js

```diff
@@ -56,21 +56,26 @@ foam.CLASS({
           };
 
           context.foam.RELATIONSHIP = function(r) {
-            var references = foam.json.references(x, r);
-
-            Promise.all(references.concat([
-              foam.package.waitForClass(r.sourceModel),
-              foam.package.waitForClass(r.targetModel)
-            ])).then(function() {
-              var obj = foam.dao.Relationship.create(r, x);
-
+            relationship = Promise.all(foam.json.references(x, r)).then(function() {
```

Comment:
Can we clean this up further by making context.foam.RELATIONSHIP match context.foam.ENUM and context.foam.INTERFACE in behaviour?

We might need to be careful about how we compute the id of the json, and maybe special case the logic of self.cache so that we don't cache a relationship but init it eagerly like you do here.